ui-frontend/packages/catalog-ui-search/src/main/webapp/component/tabs/metacard/summary.tsx

```diff
@@ -360,34 +387,24 @@ export const Editor = ({
             } catch (err) {
               console.log(err)
             }
-            const payload = [
-              {
-                ids: [lazyResult.plain.metacard.properties.id],
-                attributes: [
-                  {
-                    attribute: attr,
-                    values: transformedValues,
-                  },
-                ],
-              },
-            ]
-            setTimeout(() => {
-              $.ajax({
-                url: `./internal/metacards?storeId=${lazyResult.plain.metacard.properties['source-id']}`,
-                type: 'PATCH',
-                data: JSON.stringify(payload),
-                contentType: 'application/json',
-              })
-                .then((response: any) => {
-                  ResultUtils.updateResults(lazyResult.getBackbone(), response)
-                })
-                .always(() => {
-                  setTimeout(() => {
-                    addSnack('Successfully updated.')
-                    onSave()
-                  }, 1000)
-                })
-            }, 1000)
+
+            const attributes = [{ attribute: attr, values: transformedValues }]
+
+            const onComplete = () =>
+              setTimeout(() => {
+                addSnack('Successfully updated.')
+                onSave()
+              }, 1000)
+
+            if (ExtensionPoints.handleMetacardUpdate) {
+              ExtensionPoints.handleMetacardUpdate(
+                lazyResult,
+                attributes,
+                onComplete
```

Comment:
Just a thought (don't know if this will work or if it's any better than what you have currently), but maybe we could have the handleMetacardUpdate function return a promise and call the onComplete afterwards upstream instead?

Reply:
That's a good thought. I'll take a look here in a bit.